gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,302 @@
+
+    # Filter the  QC MT to the specified pop
+    pop_mt = mt.filter_cols(mt.population_inference.pop == pop)
+
+    # Remove hard filtered samples
+    pop_mt = pop_mt.filter_cols(~pop_mt.sample_filters.hard_filtered)
+
+    pop_mt = pop_mt.filter_rows(hl.agg.any(pop_mt.GT.is_non_ref()))
+
+    # Generate a QC MT for the given pop
+    pop_qc_mt = get_qc_mt(
+        pop_mt,
+        min_af=min_af,
+        min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
+        min_hardy_weinberg_threshold=min_hardy_weinberg_threshold,
+        ld_r2=ld_r2,
+    )
+
+    return pop_qc_mt
+
+
+def annotate_subpop_meta(ht: hl.Table) -> hl.Table:
+    meta_ht = meta.ht()
+    ht = ht.annotate(**meta_ht[ht.key])
+    ht = ht.annotate(
+        subpop_description=ht.project_meta.subpop_description,
+        v2_pop=ht.project_meta.v2_pop,
+        v2_subpop=ht.project_meta.v2_subpop,
+        pop=ht.population_inference.pop,
+        project_id=ht.project_meta.project_id,
+        project_pop=ht.project_meta.project_pop,
+    )
+
+    return ht
+
+
+def main(args):  # noqa: D103
+    pop = args.pop
+    include_unreleasable_samples = args.include_unreleasable_samples
+    high_quality = args.high_quality
+    # Read in the raw mt
+    mt = get_gnomad_v3_mt(key_by_locus_and_alleles=True)
+
+    # Filter to test partitions if specified
+    if args.test:
+        logger.info("Filtering to the first two partitions of the MT")
+        mt = mt._filter_partitions(range(2))
+
+    # Write out the densified MT
+    if args.make_full_subpop_qc_mt:
+        logger.info("Generating densified MT to use for all subpop analyses...")
+        mt = compute_subpop_qc_mt(mt, args.min_popmax_af)
+        mt = mt.checkpoint(subpop_qc.path, overwrite=args.overwrite)
+
+    if args.run_subpop_pca:
+        logger.info("Filtering subpop QC MT...")
+        mt = subpop_qc.mt()
+        mt = filter_subpop_qc(
+            mt,
+            pop,
+            args.min_af,
+            args.min_inbreeding_coeff_threshold,
+            args.min_hardy_weinberg_threshold,
+            args.ld_r2,
+            args.include_unreleasable_samples,
+        )
+
+        if args.checkpoint_filtered_subpop_qc:
+            mt = mt.checkpoint(filtered_subpop_qc_mt(pop), overwrite=args.overwrite)
+
+        if not include_unreleasable_samples:
+            mt = mt.filter_cols(~mt.project_meta.releasable | mt.project_meta.exclude)
+        if high_quality:
+            mt = mt.filter_cols(mt.high_quality)
+        if args.outlier_ht_path is not None:
+            outliers = hl.read_table(outlier_ht_path)
+            mt = mt.filter_cols(hl.is_missing(outliers[mt.col_key]))
+
+        logger.info("Generating PCs for subpops...")
+        relateds = pca_related_samples_to_drop.ht()
+        pop_pca_evals, pop_pca_scores, pop_pca_loadings = run_pca_with_relateds(
+            mt, relateds
+        )
+
+        pop_pca_evals = pop_pca_evals.checkpoint(
```

Comment:
Let's just write these out
```suggestion
        pop_pca_evals_ht = hl.Table.parallelize(
            hl.literal(
                [
                    {"PC": i + 1, "eigenvalue": x}
                    for i, x in enumerate(pop_pca_evals)
                ],
                "array<struct{PC: int, eigenvalue: float}>",
            )
        )
        pop_pca_evals_ht.write(
```